agent/proxy/manager.go

```diff
@@ -434,7 +437,7 @@ func (m *Manager) newProxy(mp *local.ManagedProxy) (Proxy, error) {
 		}
 
 		// Pass in the environmental variables for the proxy process
-		cmd.Env = os.Environ()
+		cmd.Env = append(os.Environ(), m.ProxyEnv...)
```

Comment:
Should we de-dupe this? I'm not sure what the behaviour is if any of the CONSUL_* vars are already set in agent's ENV across different platforms.

I feel like we should allow the agent's own ENV vars to _take precedence_. In general if they are set they are likely set to same thing anyway but it provides a nice way to be able to override the choice of which protocol/listener to pass which we could document. The edge case of some legacy garbage value in ENV being used seems like a reasonable trade.

Reply:
As for de-dupe it should be unnecessary. With the current code the ProxyEnv vars will be used. However we could just do it like this `append(m.ProxyEnv, os.Environ()...)` and then the agents env will be used.

Reply:
Another note is that regardless of the env we set they can still use a custom proxy command like `env CONSUL_HTTP_ADDR=.... <path to consul> connect proxy` to override anything we set.